zipper.py

```diff
@@ -1,118 +1,139 @@
+
+
+def define_zip_info(path: str, timestamp: int = MIN_EPOCH, strip_prefix: str = None):
+    if strip_prefix is not None:
+        path = os.path.relpath(path, strip_prefix)
+
+    if timestamp < MIN_EPOCH:  # prevent errors with zip dated before 1980
+        dt = MIN_EPOCH
+    else:
+        dt = convert_epoch_to_datetime_parts(timestamp)
+
+    info = ZipInfo(filename=path, date_time=dt)
+    info.external_attr = 0o777 << 16  # give full access to included file
+    info.compress_type = ZIP_DEFLATED
+
+    return info
+
+
+def bundle_files(
+    files: List[str],
+    output_file: str,
+    entrypoint: str,
+    strip_prefix: str,
+    path_vars: str,
+    timestamp: str = MIN_EPOCH,
+):
+    f"""Bundle files into a zip file.
+
+    Args:
+        files (List[str]): Files to bundle into a zip file.
+        output_file (str): Output zip filepath.
+        entrypoint (str): The entrypoint for the zip file.
+        strip_prefix (str): The root of the executable.
+        path_vars (str): Comma delimited list of extra paths.
+        timestamp (str): The unix time to use for files added into the zip.
+            Values prior to Jan 1, 1980 are ignored. Defaults to {MIN_EPOCH}.
+    """
+    # TODO: Assert entrypoint in listed files
+    preamble = gen_path_loader(path_vars, strip_prefix)
+
+    with ZipFile(output_file, "w") as ofile:
+        for f in files:
+            with open(f, "rb") as ifile:
+                script = ifile.read()
+            if f == entrypoint:
+                script = preamble.encode("utf-8") + b"\n" + script
+                info = define_zip_info(f, timestamp, strip_prefix)
+            else:
+                info = define_zip_info(f, timestamp)
+
+            ofile.writestr(info, script)
+
+
+def main(args):
+    bundle_files(**args.__dict__)
+
+
+if __name__ == "__main__":
+    import argparse
+
     parser = argparse.ArgumentParser(
-        description='create a zip file', fromfile_prefix_chars='@'
+        description="create a zip file", fromfile_prefix_chars="@"
     )
 
     parser.add_argument(
-        '-o', '--output', type=str, help='The output zip file path.'
+        "-o", "--output_file", type=str, help="The output zip file path."
     )
-
     parser.add_argument(
-        '-e', '--entrypoint', type=str, help='The entrypoint for the function',
+        "-e", "--entrypoint", type=str, help="The entrypoint for the function"
     )
-
     parser.add_argument(
-        '-t',
-        '--timestamp',
+        "-t",
+        "--timestamp",
         type=int,
-        default=ZIP_EPOCH,
-        help='The unix time to use for files added into the zip. values prior to'
-        ' Jan 1, 1980 are ignored.',
+        default=MIN_EPOCH,
+        help=(
+            "The unix time to use for files added into the zip. Values prior "
+            "to 1980-01-01 00:00:00 UTC are ignored."
+        ),
     )
-
     parser.add_argument(
-        '-s',
-        '--stripprefix',
+        "-s",
```

Comment:
Just out of curiosity, are you using "black" for formatting here?

Reply:
I'm using black, yeah.